### DISCUSSION TITLE
Spin up VS Code Server in Build

### DISCUSSION THREAD

**Discussion by carloscabanero** [blinksh/blink#1727] [Show and tell]
A place for discussing how to [use VS Code Server as an editor in Build](https://github.com/blinksh/blink/discussions/docs.blink.sh/build-up/code_server).

**Comment by steverhall** [comment]
In the intro video for install code server, it says to copy the installed server to the user’s /bin directory. It appears to work fine without doing so. What is the reason for copying to ~/bin?

**Reply by carloscabanero** [reply]
Build provides two disks, one is ephemeral and mounted on root, and the other is persistent and mounted on $HOME. The script installs VS Code in /usr/bin, which falls into the ephemeral part of the container. By moving it to ~/bin, I make sure I always have it available.

Long read for Storage in Build is here: https://docs.blink.sh/build/cloud_disk

We cannot bundle it inside our container image because of licensing issues. And we do not want to bundle any of the other "open source" versions bc then you do not have access to the Marketplace. So this is a good in-between solution. There may be a different way, so any ideas are welcome.

**Reply by steverhall** [reply]
Okay, it's specific to your Build service, not to self-hosted code-server. Thanks!

**Comment by jasonmcewen** [comment]
Trying to get latex going in a code-server.  

1. I successfully set up a code server instance using the instructions [here](https://docs.blink.sh/build-up/code_server). 
2. Also installed latex by running:
``` 
sudo apt install texlive texlive-science texlive-latex-extra texlive-pstricks latexmk
```
3. Installed [Latex Workshop Extension](https://marketplace.visualstudio.com/items?itemName=James-Yu.latex-workshop).
4. Cloned a repo that contains latex files.  Successfully built pdf from latex source (CMD + Option + B).  

However, when I try to view the built pdf (either opening the pdf in VSCode or by CMD + Option + V), an empty window is opened but the pdf is not rendered.

**Reply by carloscabanero** [reply]
Haven't tried this myself, but it looks like a possible issue with VS Code itself. We still do not connect the Files.app with Blink Build, but some users in the meantime have installed Dropbox sync on Build directly, as a way to sync files outside, even after the machine is fully down.

**Reply by jasonmcewen** [reply]
Thanks @carloscabanero.  Managed to get this working on GitHub codespaces so using that for now.  Would be great to do this on a Blink build machine, however, since that would provide more flexibility.

**Reply by carloscabanero** [reply]
No worries, latex is an important flow for us, so we will improve things. I would like to add more latex tools in the default image. Are those you installed the usual?

**Reply by jasonmcewen** [reply]
That would be great.  Those latex packages are fairly typical I guess, particularly for writing papers.  It's prehaps a little more than strictly necessary.  

For example `latexmk` may not be so widely used.  However, the VSCode Latex Workshop extension does use it so if people what to use VSCode then they would probably want it.

**Comment by eurobob** [comment]
I get the following error when trying to run "code-server" after moving to ~/bin

/root/bin/code-server: line 1: syntax error near unexpected token 'newline'
/root/bin/code-server: line 1: '‹!DOCTYPE html>'

**Reply by carloscabanero** [reply]
I think this should go to a different Discussion as you are using code-server instead of VS Code by Microsoft, right? I haven't tried code-serve in a while, but is there any reason why you are using that one?

**Reply by eurobob** [reply]
@carloscabanero no particular reason, i'm just trying to figure things out and I think I was following old docs.

I use `code tunnel` now and that works fine. Not sure where I found that recommendation though 🤣

**Comment by SidneyMasters** [comment]
> A place for discussing how to [use VS Code Server as an editor in Build](https://github.com/blinksh/blink/discussions/docs.blink.sh/build-up/code_server).

Hi Carlos,

First of all please forgive me as I im new to Blink Build so some of my questions might have been answered elsewere. The link above does not lead anywhere therefore I am placing my question here. After i install VS Code-server using your video, I get the foloowing error :
/root/bin/code-server: line 1: syntax error near unexpected token `newline'
/root/bin/code-server: line 1: `<!DOCTYPE html>'

What am I doing wrong 

Thank you

**Reply by carloscabanero** [reply]
Hi! Sorry the instructions are a bit old and we have not updated the article yet. Please check this out as this is what we recommend now: https://www.reddit.com/r/BlinkShell/comments/13vrajt/comment/jmbdbpp/?utm_source=share&utm_medium=web2x&context=3
